express-zod-api/src/documentation.ts

```diff
@@ -94,22 +85,18 @@ interface DocumentationParams {
 export class Documentation extends OpenApiBuilder {
   readonly #lastSecuritySchemaIds = new Map<SecuritySchemeType, number>();
   readonly #lastOperationIdSuffixes = new Map<string, number>();
-  readonly #references = new Map<z.ZodTypeAny, string>();
+  readonly #references = new Map<object, string>();
 
   #makeRef(
-    schema: z.ZodTypeAny,
-    subject:
-      | SchemaObject
-      | ReferenceObject
-      | (() => SchemaObject | ReferenceObject),
-    name = this.#references.get(schema),
+    key: object,
+    subject: SchemaObject | ReferenceObject,
+    name = this.#references.get(key),
   ): ReferenceObject {
     if (!name) {
-      name = `Schema${this.#references.size + 1}`;
-      this.#references.set(schema, name);
-      if (typeof subject === "function") subject = subject();
+      name = `Schema${Object.keys(this.rootDoc.components?.schemas || {}).length + 1}`;
```

Comment:
can use `this.#refereces.size` here
```suggestion
      name = `Schema${this.#references.size + 1}`;
```